tests/test_observables.py

```diff
@@ -268,6 +269,16 @@ def test_tor(self, mock_get=None, mock_post=None):
         ).start()
         self.assertEqual(report.get("success", False), True)
 
+    def test_firehol_iplist(self, mock_get=None, mock_post=None):
+        report = firehol_iplist.FireHol_IPList(
+            "FireHol_IPList",
+            self.job_id,
+            self.observable_name,
+            self.observable_classification,
+            {},
```

Comment:
there is a hidden problem here. Considering that `list_names` is initialized as empty, this would mean that most of the code is not reached by tests here. Could you add the default list `firehol_level1.netset` here or, maybe better, directly in the `set_config` method? https://github.com/intelowlproject/IntelOwl/pull/368/files#diff-3b52a5391525c883aaa2861e5aef8d6fea9e50cb88f8555e80a137ce40437938R19

Reply:
That's a beautiful catch! Adding it asap.

Reply:
If the key `list_names` exists in the `analyzer_config.py` and it is empty, it would override the default configuration.

Yes, your point is good, we could add an `AnalyzerConfigurationException` in the case the `list_names` is empty to manage also that case. Could you do it? Then we can merge

Reply:
Okay! I added `AnalyzerConfigurationException` for the empty list case. While testing I noticed a `522 Server Error` and the final error that showed up in the application was:
![image](https://user-images.githubusercontent.com/28859415/113698266-e9e02e00-96f1-11eb-8b0f-6b832076e468.png)
Since, it didn't download the db, how should we handle this error?

Reply:
the better things should be to mock the request to FireHol so we do not really contact the server in the tests. You can check how `test_triage_search` is implemented and try to add something similar.

Reply:
I couldn't explain it properly, my bad. I ran a scan in the web application to check if the analyzer was working as expected and got that error, so, I was asking if we can/should handle this? This is the whole traceback:

```
intelowl_celery_worker_default | [2021-04-06 10:30:04,459: INFO/MainProcess] starting download of firehol_level1.netset from firehol iplist
intelowl_celery_worker_default | Traceback (most recent call last):
intelowl_celery_worker_default |   File "/opt/deploy/intel_owl/api_app/script_analyzers/observable_analyzers/firehol_iplist.py", line 60, in download_iplist
intelowl_celery_worker_default |     r.raise_for_status()
intelowl_celery_worker_default |   File "/usr/local/lib/python3.8/site-packages/requests/models.py", line 943, in raise_for_status
intelowl_celery_worker_default |     raise HTTPError(http_error_msg, response=self)
intelowl_celery_worker_default | requests.exceptions.HTTPError: 522 Server Error:  for url: https://iplists.firehol.org/files/firehol_level1.netset
intelowl_celery_worker_default | [2021-04-06 10:30:37,763: ERROR/MainProcess] 522 Server Error:  for url: https://iplists.firehol.org/files/firehol_level1.netset
intelowl_celery_worker_default | Traceback (most recent call last):
intelowl_celery_worker_default |   File "/opt/deploy/intel_owl/api_app/script_analyzers/observable_analyzers/firehol_iplist.py", line 60, in download_iplist
intelowl_celery_worker_default |     r.raise_for_status()
intelowl_celery_worker_default |   File "/usr/local/lib/python3.8/site-packages/requests/models.py", line 943, in raise_for_status
intelowl_celery_worker_default |     raise HTTPError(http_error_msg, response=self)
intelowl_celery_worker_default | requests.exceptions.HTTPError: 522 Server Error:  for url: https://iplists.firehol.org/files/firehol_level1.netset
intelowl_celery_worker_default | Traceback (most recent call last):
intelowl_celery_worker_default |   File "/opt/deploy/intel_owl/api_app/script_analyzers/classes.py", line 99, in start
intelowl_celery_worker_default |     result = self.run()
intelowl_celery_worker_default |   File "/opt/deploy/intel_owl/api_app/script_analyzers/observable_analyzers/firehol_iplist.py", line 32, in run
intelowl_celery_worker_default |     self.check_iplist_status(list_name)
intelowl_celery_worker_default |   File "/opt/deploy/intel_owl/api_app/script_analyzers/observable_analyzers/firehol_iplist.py", line 87, in check_iplist_status
intelowl_celery_worker_default |     timestamp = os.path.getctime(iplist_location)
intelowl_celery_worker_default |   File "/usr/local/lib/python3.8/genericpath.py", line 65, in getctime
intelowl_celery_worker_default |     return os.stat(filename).st_ctime
intelowl_celery_worker_default | FileNotFoundError: [Errno 2] No such file or directory: '/opt/deploy/files_required/firehol_level1.netset'
intelowl_celery_worker_default | [2021-04-06 10:30:37,764: ERROR/MainProcess] job_id:3, analyzer:'FireHol_IPList'. Unexpected error: '[Errno 2] No such file or directory: '/opt/deploy/files_required/firehol_level1.netset''
intelowl_celery_worker_default | Traceback (most recent call last):
intelowl_celery_worker_default |   File "/opt/deploy/intel_owl/api_app/script_analyzers/classes.py", line 99, in start
intelowl_celery_worker_default |     result = self.run()
intelowl_celery_worker_default |   File "/opt/deploy/intel_owl/api_app/script_analyzers/observable_analyzers/firehol_iplist.py", line 32, in run
intelowl_celery_worker_default |     self.check_iplist_status(list_name)
intelowl_celery_worker_default |   File "/opt/deploy/intel_owl/api_app/script_analyzers/observable_analyzers/firehol_iplist.py", line 87, in check_iplist_status
intelowl_celery_worker_default |     timestamp = os.path.getctime(iplist_location)
intelowl_celery_worker_default |   File "/usr/local/lib/python3.8/genericpath.py", line 65, in getctime
intelowl_celery_worker_default |     return os.stat(filename).st_ctime
intelowl_celery_worker_default | FileNotFoundError: [Errno 2] No such file or directory: '/opt/deploy/files_required/firehol_level1.netset'
```

Yes, you are right we should mock the request for testing, I will try to implement it!